src/data_readers/data_reader_jag_conduit.cpp

```diff
@@ -789,7 +871,7 @@ std::vector<data_reader_jag_conduit::scalar_t> data_reader_jag_conduit::get_scal
     const conduit::Node & n_scalar = get_conduit_node(scalar_key);
```

Comment:
FYI, I think the data_store version will be:
const char* data_store_jag_conduit::get_scalar(std::string scalar_key);
  - or possibly -
std::pair<const char*, uint> data_store_jag_conduit::get_scalar(std::string scalar_key);
where 'uint' is the number of bytes.
I'm thinking that data_store will return all values as const char*, which the data_reader can then cast them as desired. Let me know if you think this won't work, or something else works better.

Reply:
I see. It seems like there is no need for having separate get_scalar() and get_input().
We may be able to use std::pair<const char*, TypeID> instead. In which case, you need some special handling of CHAR8_STR_ID.